ctrl.c

```diff
@@ -924,25 +924,35 @@ static struct pv_cmd *pv_ctrl_process_endpoint_and_reply(
 			if (pv_ctrl_process_put_file(req_fd, content_length,
 						     file_path_tmp) < 0)
 				goto out;
-			if (pv_storage_validate_file_checksum(file_path_tmp,
-							      file_name) < 0) {
+			if (pv_fs_path_exist(file_path) &&
```

Comment:
shouldn't we do this before pv_ctrl_process_put_file and return a 409 in case it exists and checksum is good? if not good we can remove and proceed with pv_ctrl_process_put_file. That way we would avoid unnecessary file transfers

Reply:
sure thought about it, but dont want to change behaviour .... there are clients written against this that then suddenly start erroring etc.... so I chose not to do that.